File: Components/DevExpressVCL/Demos/ExpressEditors Library/CBuilder/BaseForm.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "BaseForm.h"
#include "AboutDemoForm.h"
#include "DemoUtils.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxControls"
#pragma link "cxClasses"
#pragma link "cxLookAndFeels"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma resource "*.dfm"
//---------------------------------------------------------------------------
__fastcall TfmBaseForm::TfmBaseForm(TComponent* Owner): TForm(Owner)
{
	BuildLookAndFeelMenu();
};
//---------------------------------------------------------------------------
void __fastcall TfmBaseForm::BuildLookAndFeelMenu()
{
	mmMain->Items->Insert(mmMain->Items->IndexOf(miAbout),
		CreateLookAndFeelMenuItems(mmMain->Items, NULL, 1));
}
//---------------------------------------------------------------------------
void __fastcall TfmBaseForm::miAboutClick(TObject *Sender)
{
	ShowAboutDemoForm();
}
//---------------------------------------------------------------------------
void __fastcall TfmBaseForm::miExitClick(TObject *Sender)
{
	Close();
}
//---------------------------------------------------------------------------

File: Components/DevExpressVCL/Demos/ExpressEditors Library/Delphi/BaseForm.pas
unit BaseForm;

{$I cxVer.inc}

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, Menus, ComCtrls, dxCore, cxClasses, cxLookAndFeels;

type
  TfmBaseForm = class(TForm)
    lbDescription: TLabel;
    miAbout: TMenuItem;
    miExit: TMenuItem;
    miFile: TMenuItem;
    mmMain: TMainMenu;
    procedure miAboutClick(Sender: TObject);
    procedure miExitClick(Sender: TObject);
  protected
    procedure BuildLookAndFeelMenu; virtual;
  public
    procedure AfterConstruction; override;
  end;

var
  fmBaseForm: TfmBaseForm;

implementation

{$R *.dfm}

uses
  AboutDemoForm, DemoUtils, ActnList;

{ TfmBaseForm }

procedure TfmBaseForm.AfterConstruction;
begin
  BuildLookAndFeelMenu;
  dxBarConvertMainMenu(mmMain);
  lbDescription.Transparent := False;
  inherited AfterConstruction;
end;

procedure TfmBaseForm.BuildLookAndFeelMenu;
begin
  mmMain.Items.Insert(mmMain.Items.IndexOf(miAbout), CreateLookAndFeelMenuItems(mmMain.Items
    {$IFNDEF EXPRESSBARS}, TActionList.Create(Self) {$ENDIF}));
end;

procedure TfmBaseForm.miAboutClick(Sender: TObject);
begin
  ShowAboutDemoForm;
end;

procedure TfmBaseForm.miExitClick(Sender: TObject);
begin
  Close;
end;

end.
